tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/update_invalid_zip_codes_test.py

```diff
@@ -0,0 +1,159 @@
+        # mocked for this test
+        cls.zip3_lookup_name = f'{cls.project_id}.{cls.dataset_id}.{ZIP3_LOOKUP}'
+
+        cls.fq_table_names = [
+            f"{project_id}.{dataset_id}.{OBSERVATION}", cls.zip3_lookup_name
+        ]
+        # call super to set up the client, create datasets, and create
+        # empty test tables
+        # NOTE:  does not create empty sandbox tables.
+        super().setUpClass()
+
+    def setUp(self):
+        """
+        Create common information for tests.
+
+        Creates common expected parameter types from cleaned tables and a common fully qualified (fq)
+        dataset name string used to load the data.
+        """
+        fq_dataset_name = self.fq_table_names[0].split('.')
+        self.fq_dataset_name = '.'.join(fq_dataset_name[:-1])
+
+        self.date = parser.parse('2019-03-03').date()
+        self.invalid_zip_response = 'Response removed due to invalid value'
+        self.invalid_zip_concept = 2000000010
+
+        super().setUp()
+
+    def test_update_invalid_zip_codes(self):
+        """
+        Tests that the specifications for SANDBOX_INVALID_ZIP_CODES and UPDATE_INVALID_ZIP_CODES perform as designed.
+
+        Validates pre conditions, tests execution, and post conditions based on the load
+        statements and the tables_and_counts variable.
+        """
+
+        zip3_lookup_tmpl = self.jinja_env.from_string("""
+            INSERT INTO `{{fq_dataset_name}}.zip3_lookup` (zip3)
+            VALUES
+                (123), (234)
+            """).render(fq_dataset_name=self.fq_dataset_name)
+
+        observation_tmpl = self.jinja_env.from_string("""
+            INSERT INTO `{{fq_dataset_name}}.observation` (observation_id, person_id, observation_concept_id, 
+                observation_date, observation_type_concept_id, observation_source_concept_id, value_as_number, 
+                value_as_string, value_source_concept_id, value_source_value)
+            VALUES
+                -- valid zip codes, will not be affected --
+                (1, 1, 0, date('2019-03-03'), 0, 1585250, null, '12345', null, null), 
+                (2, 2, 0, date('2019-03-03'), 0, 1585250, null, '23456-1234', null, null),
+                
+                -- invalid zip codes, will be sandboxed and updated in observation table --
+                (3, 3, 0, date('2019-03-03'), 0, 1585250, null, '1234', null, null), 
+                (4, 4, 0, date('2019-03-03'), 0, 1585250, null, '123', null, null),
+                (5, 5, 0, date('2019-03-03'), 0, 1585250, null, '12', null, null),
+                (6, 6, 0, date('2019-03-03'), 0, 1585250, null, '1', null, null),
+                
+                -- not zip codes (not zip code concept id), will not be affected --
+                (7, 7, 0, date('2019-03-03'), 0, 1111111, null, '111', null, null),
+                (8, 8, 0, date('2019-03-03'), 0, 2222222, null, '222', null, null)
```

Comment:
I added these extra test scenarios corresponding to the cases mentioned in the cleaning rule feedback. [No need to commit unless you want to]
```suggestion
                (8, 8, 0, date('2019-03-03'), 0, 2222222, null, '222', null, null),
                
                -- invalid zip codes with 5 characters and unknown zip 3 --
                (9, 9, 0, date('2019-03-03'), 0, 1585250, null, '00501', null, null),

                -- invalid zip codes with 5 characters, last 2 characters non-numeric --
                (10, 10, 0, date('2019-03-03'), 0, 1585250, null, '123ab', null, null),

                -- valid zip codes with 5 numeric characters but extra whitespace --
                (11, 11, 0, date('2019-03-03'), 0, 1585250, null, '    12345 ', null, null)                
```